File: app/Services/Reports/DashboardReport.php
<?php

namespace App\Services\Reports;

use App\Enums\BookingStatus;
use App\Enums\BusinessStatus;
use App\Models\Booking;
use App\Models\Customer;
use App\Models\Staff;
use App\Models\Tenant;
use App\Support\DateRange;
use Carbon\CarbonImmutable;

/**
 * The Reports section's "Overview" — a date-range-flexible superset of the
 * fixed-to-today `GET /dashboard/summary` (Phase 8, still used by the Home
 * tab; see DashboardController). Composes `BookingReport`/`RevenueReport`
 * rather than re-querying, so the two endpoints' numbers can never drift
 * apart for the same range/filters.
 */
class DashboardReport
{
    public function __construct(
        private readonly BookingReport $bookings,
        private readonly RevenueReport $revenue,
    ) {}

    /**
     * @param  array<string, mixed>  $filters
     */
    public function generate(Tenant $tenant, DateRange $range, array $filters): array
    {
        $bookingSummary = $this->bookings->generate($tenant, $range, $filters)['summary'];
        $revenueSummary = $this->revenue->generate($tenant, $range, $filters)['summary'];

        $today = CarbonImmutable::now($range->timezone)->toDateString();
        $nextAppointmentQuery = Booking::query()->with('customer')
            ->whereIn('status', array_map(fn (BookingStatus $s) => $s->value, [
                BookingStatus::PENDING, BookingStatus::CONFIRMED, BookingStatus::CHECKED_IN, BookingStatus::IN_SERVICE,
            ]))
            ->where('booking_date', '>=', $today);
        if (! empty($filters['branch_id'])) {
            $nextAppointmentQuery->where('branch_id', $filters['branch_id']);
        }
        $nextAppointment = $nextAppointmentQuery->orderBy('booking_date')->orderBy('start_time')->first();

        return [
            'summary' => [
                'range' => ['preset' => $range->preset, 'from' => $range->startDate(), 'to' => $range->endDate()],
                'bookings' => $bookingSummary,
                'revenue' => $revenueSummary,
                'active_staff' => Staff::query()->where('status', BusinessStatus::ACTIVE)->count(),
                'total_customers' => Customer::query()->count(),
                'next_appointment' => $nextAppointment ? [
                    'id' => $nextAppointment->id,
                    'booking_date' => $nextAppointment->booking_date->format('Y-m-d'),
                    'start_time' => substr((string) $nextAppointment->start_time, 0, 5),
                    'status' => $nextAppointment->status->value,
                    'customer_name' => $nextAppointment->customer?->name,
                ] : null,
            ],
        ];
    }
}

File: mobile/lib/features/owner/reports/data/models/dashboard_report.dart
/// Mirrors `App\Services\Reports\DashboardReport` — the Reports section's
/// date-range-flexible overview (`GET /reports/dashboard`). Distinct from
/// `DashboardSummary` (Phase 8's fixed-to-today Home tab widget).
class DashboardReportSummary {
  const DashboardReportSummary({
    required this.rangeFrom,
    required this.rangeTo,
    required this.bookings,
    required this.revenue,
    required this.activeStaff,
    required this.totalCustomers,
    this.nextAppointment,
  });

  factory DashboardReportSummary.fromJson(Map<String, dynamic> json) {
    final summary = json['summary'] as Map<String, dynamic>;
    final range = summary['range'] as Map<String, dynamic>;
    return DashboardReportSummary(
      rangeFrom: range['from'] as String,
      rangeTo: range['to'] as String,
      bookings: Map<String, dynamic>.from(summary['bookings'] as Map),
      revenue: Map<String, dynamic>.from(summary['revenue'] as Map),
      activeStaff: summary['active_staff'] as int,
      totalCustomers: summary['total_customers'] as int,
      nextAppointment: summary['next_appointment'] as Map<String, dynamic>?,
    );
  }

  final String rangeFrom;
  final String rangeTo;
  final Map<String, dynamic> bookings;
  final Map<String, dynamic> revenue;
  final int activeStaff;
  final int totalCustomers;
  final Map<String, dynamic>? nextAppointment;

  int bookingCount(String status) => (bookings[status] as num?)?.toInt() ?? 0;
  String get netRevenue => revenue['net_revenue']?.toString() ?? '0.00';
}
